### PR TITLE
GH-2483 Convert Dashy to a Widget

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes GH-2483

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage please review - the important bits are in packages/marshaller

File: apps/legacyDash/src/app.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ const logger = scopedLogger("app");
 export class App extends DockPanel {
     private _url = new Comms.ESPUrl().url(document.URL);
     private _ddlv1 = new JSONEditor();
+    private _ddlv2XXX = new JSONEditor();
```
Comment (jbrundage) [inline apps/legacyDash/src/app.ts]: What is this compared to `_ddlv2`? Maybe `_ddlv2XXX` should be named more appropriately?

File: demos/dashy/src/dashComms.ts [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
-        newUrl = btoa(`${opts.baseUrl}/${action}?${serializeRequest(request)}`);
-    }
-    const connection = new Connection({ baseUrl: "https://webmiscdev.risk.regn.net", type: "get" });
-    return connection.send("brundajx/DASH2/demos/dashy/bis_proxy.php", { encoded: newUrl }, responseType).then(response => {
-        return response;
-    }).catch(e => {
-        throw e;
+export function hookComms() {
+    const origSend = hookSend((opts: IOptions, action: string, request: any, responseType: ResponseType): Promise<any> => {
+        if (opts.baseUrl === "https://webmiscdev.risk.regn.net") {
```
Comment (jbrundage) [inline demos/dashy/src/dashComms.ts]: These references to `webmiscdev.risk.regn.net` should either be removed or replaced with `webmiscdev01.risk.regn.net`

File: demos/dashy/src/dashComms.ts [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
+            return origSend(opts, action, request, responseType);
+        }
+        let newUrl = "";
+        if (opts.baseUrl.split("").reverse()[0] === "/" || action[0] === "/") {
+            newUrl = btoa(`${opts.baseUrl}${action}?${serializeRequest(request)}`);
+        } else {
+            newUrl = btoa(`${opts.baseUrl}/${action}?${serializeRequest(request)}`);
+        }
+        const connection = new Connection({ baseUrl: "https://webmiscdev.risk.regn.net", type: "get" });
```
Comment (jbrundage) [inline demos/dashy/src/dashComms.ts]: These references to `webmiscdev.risk.regn.net` should either be removed or replaced with `webmiscdev01.risk.regn.net`

**Pull request merged**